examples/simple_daplink/simple_daplink.ino

```diff
@@ -92,213 +92,22 @@ uint8_t const desc_hid_report[] =
-    return -1;
-  } else if ((len == 4) && ((codepoint < 0x010000) || (codepoint > 0x10FFFF))) {
-    // "You might expect larger code points than U+10FFFF
-    // to be expressible, but Unicode is limited in Sections 12
-    // of RFC3629 to match the limits of UTF-16." -- Wikipedia UTF-8 note
-    // See https://tools.ietf.org/html/rfc3629#section-12
-    return -1;
-  }
-
-  // high and low surrogate halves (U+D800 through U+DFFF) used by UTF-16 are
-  // not legal Unicode values ... see RFC 3629.
-  if ((codepoint >= CODEPOINT_LOWEST_SURROGATE_HALF) && (codepoint <= CODEPOINT_HIGHEST_SURROGATE_HALF)) {
-    return -1;
-  }
-
-  *codepointp = codepoint;
-  return len;
-}
-
-
-static int strcpy_utf16(const char *s, uint16_t *buf, int bufsize)
-{
-  int i = 0;
-  int buflen = 0;
-
-  while (s[i] != 0) {
-    uint32_t codepoint;
-    int8_t utf8len = utf8Codepoint((const uint8_t *)s + i, &codepoint);
-
-    if (utf8len < 0) {
-      // Invalid utf8 sequence, skip it
-      i++;
-      continue;
-    }
-
-    i += utf8len;
-
-    if (codepoint <= 0xffff) {
-      if (buflen == bufsize)
-        break;
-
-      buf[buflen++] = codepoint;
-
-    } else {
-      if (buflen + 1 >= bufsize)
-        break;
-
-      // Surrogate pair
-      codepoint -= 0x10000;
-      buf[buflen++] = (codepoint >> 10) + 0xd800;
-      buf[buflen++] = (codepoint & 0x3ff) + 0xdc00;
-    }
-  }
-
-  return buflen;
-}
-
-
-// Invoked when received GET STRING DESCRIPTOR request
-// Application return pointer to descriptor, whose contents must exist long enough for transfer to complete
-// Note: the 0xEE index string is a Microsoft OS 1.0 Descriptors.
-// https://docs.microsoft.com/en-us/windows-hardware/drivers/usbcon/microsoft-defined-usb-descriptors
-const uint16_t * tud_descriptor_string_cb(uint8_t index, uint16_t langid)
-{
-  (void) langid;
-
-  uint8_t chr_count;
-
-  switch (index)
-  {
-    case 0:
-      _desc_str[1] = USBDevice.getLanguageDescriptor();
-      chr_count = 1;
-    break;
-    case 1:
-      chr_count = strcpy_utf16(USBDevice.getManufacturerDescriptor(), _desc_str + 1, 32);
-    break;
-
-    case 2:
-      chr_count = strcpy_utf16(USBDevice.getProductDescriptor(), _desc_str + 1, 32);
-    break;
-
-    case 3:
-      // serial Number
-      chr_count = USBDevice.getSerialDescriptor(_desc_str+1);
-    break;
-    case 4:
-      // interface name
-      chr_count = strcpy_utf16("CMSIS-DAP", _desc_str + 1, 32);
-    break;
-    default: return NULL;
-  }
-
-  // first byte is length (including header), second byte is string type
-  _desc_str[0] = (TUSB_DESC_STRING << 8 ) | (2*chr_count + 2);
-
-  return _desc_str;
-}
-
-
-
-
 void setup() {
+    USBDevice.setProductDescriptor("CMSIS-DAP");
+
     usb_hid.enableOutEndpoint(true);
     usb_hid.setPollInterval(2);
     usb_hid.setBootProtocol(0);
-    USBDevice.setProductDescriptor("CMSIS-DAP");
+    usb_hid.setStringDescriptor("CMSIS-DAP");
```

Comment:
the new interface `setStringDescriptor()`

Reply:
That's good